CMakeLists.txt

```diff
@@ -1,12 +1,27 @@
-cmake_minimum_required(VERSION 3.10) #Use include_guard()
-project(CMakeTest VERSION 1.0.0)
-
-option(BUILD_TESTING "Should we build and run the unit tests?" OFF)
+set(_ct_min_cmake_version "3.14")
+cmake_minimum_required(VERSION "${_ct_min_cmake_version}") #Required for FetchContent_MakeAvailable()
+project(CMakeTest VERSION 1.0.0 LANGUAGES NONE)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}")
 set(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH}" CACHE STRING "" FORCE)
+include("${PROJECT_SOURCE_DIR}/cmake/get_cpp.cmake")
+include(cmake_test/cmake_test)
 
 if("${BUILD_TESTING}")
+    message("Building tests")
+
     include(CTest)
-    add_subdirectory(tests)
+    enable_testing()
```

Comment:
I think `include(CTest)` already calls `enable_testing()` when `BUILD_TESTING` is true.

Reply:
Huh, so it does. Was not expecting an include() to have side effects like that. I've removed that call now